contracts/P256Signer.sol

```diff
@@ -51,20 +51,41 @@ contract P256Signer {
     /// @param _hash The hash of the data signed
     /// @param _signature The signature
     /// @return The EIP-1271 magic value
-    function isValidSignature(bytes memory _hash, bytes memory _signature) public view returns (bytes4) {
+    function isValidSignature(bytes memory _hash, bytes calldata _signature) public view returns (bytes4) {
         _validate(_hash, _signature);
         return OLD_EIP1271_MAGICVALUE;
     }
 
+    struct SignatureLayout {
+        bytes authenticatorData;
+        bytes clientData;
+        uint256 challengeOffset;
+        uint256[2] rs;
+    }
+
     /// @notice Validates the signature
     /// @param data The data signed
     /// @param _signature The signature
-    function _validate(bytes memory data, bytes memory _signature) private view {
+    function _validate(bytes memory data, bytes calldata _signature) private view {
         bytes32 _hash = keccak256(data);
-        (bytes memory authenticatorData, bytes memory clientData, uint256 challengeOffset, uint256[2] memory rs) =
-            abi.decode(_signature, (bytes, bytes, uint256, uint256[2]));
-
-        bool valid = WrapperFCLWebAuthn.checkSignature(authenticatorData, 0x01, clientData, _hash, challengeOffset, rs, [x, y]);
+        SignatureLayout calldata signaturePointer;
+        // This code should precalculate the offsets of variables as defined in the layout
+        // Calldata variables are represented as offsets, and, I think, length for dynamic types
```

Comment:
> I think

Much confidence :P

Reply:
I think calldata variables are represented as offsets only when they are of dynamic size. If this is the case, I'm not sure the comment explains this well :)
I interpret the following assembly as : it "aligns" `signaturePointer` with the calldata`. Am I wrong here ?

Reply:
I think before accessing the variable and copying it to the stack with `calldataload(offset)`, it represents it as an offset. This can be seen in the Yul output I attached to another discussion